extensions/qute/runtime/src/main/java/io/quarkus/qute/runtime/devmode/QuteErrorPageSetup.java

```diff
@@ -0,0 +1,223 @@
+package io.quarkus.qute.runtime.devmode;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.stream.Collectors;
+
+import org.jboss.logging.Logger;
+
+import io.quarkus.dev.ErrorPageGenerators;
+import io.quarkus.dev.spi.HotReplacementContext;
+import io.quarkus.dev.spi.HotReplacementSetup;
+import io.quarkus.qute.Engine;
+import io.quarkus.qute.Escaper;
+import io.quarkus.qute.EvalContext;
+import io.quarkus.qute.ReflectionValueResolver;
+import io.quarkus.qute.Template;
+import io.quarkus.qute.ValueResolver;
+import io.quarkus.runtime.TemplateHtmlBuilder;
+
+public class QuteErrorPageSetup implements HotReplacementSetup {
+
+    private static final Logger LOG = Logger.getLogger(QuteErrorPageSetup.class);
+
+    private static final String TEMPLATE_EXCEPTION = "io.quarkus.qute.TemplateException";
+    private static final String ORIGIN = "io.quarkus.qute.TemplateNode$Origin";
+
+    private static final String PROBLEM_TEMPLATE = ""
+            + "<h3>#{problemIndex} {title}</h3>\n"
+            + "<div style=\"margin-bottom:0.5em;\">{description}</div>\n"
+            + "<div style=\"font-family:monospace;font-size:1em;background-color:#2E3436;color:white;padding:1em;margin-bottom:2em;\">\n"
+            + "{#if realLines.get(0) > 1}<span style=\"color:silver;\">...</span><br>{/if}\n"
+            + "{#for line in sourceLines}\n"
+            // highlight the error line - start
+            + "{#if lineNumber is realLines.get(index)}<div style=\"background-color:#555753;\">{/if}\n"
+            // line number
+            + "<span style=\"color:silver;\">{realLines.get(index).pad}</span>\n"
+            // line content
+            + " {line}\n"
+            // highlight the error line - end 
+            + "{#if lineNumber is realLines.get(index)}</div>{#else}<br>{/if}\n"
+            // point to error
+            + "{#if lineNumber is realLines.get(index)}{space.pad}<span style=\"color:red;\">{#for i in lineCharacterStart}={/for}^</span><br>{/if}\n"
+            + "{/for}\n"
+            + "{#if endLinesSkipped}<span style=\"color:silver;\">...</span>{/if}\n"
+            + "</div>";
+
+    private HotReplacementContext hotReplacementContext;
+
+    @Override
+    public void setupHotDeployment(HotReplacementContext context) {
+        this.hotReplacementContext = context;
+        ErrorPageGenerators.register(TEMPLATE_EXCEPTION, this::generatePage);
+    }
+
+    String generatePage(Throwable exception) {
+        Escaper escaper = Escaper.builder().add('"', "&quot;").add('\'', "&#39;")
```

Comment:
Doesn't this look like generally useful for pretty much anything HTML? Shouldn't we have something like `Escaper.htmlBuilder()` or `Escaper.builder().forHtml()`?

Reply:
We have an optimized version of a result mapper in the runtime module: https://github.com/quarkusio/quarkus/blob/master/extensions/qute/runtime/src/main/java/io/quarkus/qute/runtime/HtmlEscaper.java

Maybe we could replace the generic escaper with this one. I'll try to prepare a follow-up PR...